### PR TITLE
Fix write_json to handle empty string column

### PR DESCRIPTION
**Pull request by karthikeyann**
## Description
Add empty string column condition for write_json 
bypass make_strings_children for empty column because when grid size is zero, it throws cuda error.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Review by mhaseeb123 — COMMENTED** [review]
Looks good. Should we add tests for this case as well as empty columns for other types?

File: cpp/src/io/json/write_json.cu [hunk 1]
```diff
@@ -170,6 +170,8 @@ struct escape_strings_fn {
                                               rmm::cuda_stream_view stream,
                                               rmm::device_async_resource_ref mr)
   {
+    if (column_v.size() == 0)  // empty begets empty
+      return make_empty_column(type_id::STRING);
```
Comment (davidwendt) [inline cpp/src/io/json/write_json.cu:174]:
```suggestion
    if (column_v.is_empty()) { // empty begets empty
      return make_empty_column(type_id::STRING);
    }
```

**Comment by karthikeyann** [comment]
unit test added.

**Comment by karthikeyann** [comment]
>  Should we add tests for this case as well as empty columns for other types?

Not required. All other types use string functions which checks for empty inputs. Only custom function created in this file `get_escaped_strings` did not handle empty string column input. I added string, int empty column also to this unit test.

File: cpp/src/io/json/write_json.cu [hunk 2]
```diff
@@ -170,6 +170,8 @@ struct escape_strings_fn {
                                               rmm::cuda_stream_view stream,
                                               rmm::device_async_resource_ref mr)
   {
+    if (column_v.is_empty())  // empty begets empty
+      return make_empty_column(type_id::STRING);
```
Comment (davidwendt) [inline cpp/src/io/json/write_json.cu:174]:
```suggestion
    if (column_v.is_empty()) { // empty begets empty
      return make_empty_column(type_id::STRING);
    }
```

**Review by davidwendt — APPROVED** [review]
Nit on braces

**Review by mhaseeb123 — APPROVED** [review]

**Pull request merged**